### DISCUSSION TITLE
RaspAP change Authentication Password on WebGui There is no longer a login window

### DISCUSSION THREAD

**Discussion by FireDVB** [RaspAP/raspap-webgui#1522] [Q&A] [answered]
Hello my english not so good.

The RaspAP on Rpi 4B function is super with NordVPN.

I have change the password (secret) on the webgui over LAN IP and there is no longer a login window.

delete /etc/raspap/raspap.auth There is no longer a login window

WLAN is disabled
Hotspot is disabled
The router only works via LAN with NordVPN, but unfortunately there is no login window via the WebGui to enter the user and password.

Add RaspAP's speedtest server under WebGui --> Network -> Speedtest (Button)
Is that also possible?

![image](https://github.com/RaspAP/raspap-webgui/assets/125806299/ff47be22-5437-46eb-bd13-7ee8352b1bfb)
is not present on the webgui
Or is this only available when WiFi is activated?

Who has a tip for me?

Greetings FireDVB

**Comment by billz** [comment]
There are several questions here, but I'll focus on the subject of this discussion. RaspAP uses [HTTP basic authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication) to manage logins. The login window not appearing is normal behavior due to the default credentials being cached by your browser. This is expected when removing `/etc/raspap/raspap.auth` per [this FAQ](https://docs.raspap.com/faq/#password).

Regarding everything else, this appears to be a very old version of Insiders. I'd suggest upgrading.

**Reply by FranziHH** [reply]
The login no longer works since version 3.0.8. You are always! logged in.

This is a very bad idea:

in 'raspap-webgui/src/RaspAP/Auth/HTTPAuth.php'
`public function getAuthConfig()
    {
        $config = $this->auth_default;

        if (file_exists(RASPI_CONFIG . '/raspap.auth')) {
            if ($auth_details = fopen(RASPI_CONFIG . '/raspap.auth', 'r')) {
                $config['admin_user'] = trim(fgets($auth_details));
                $config['admin_pass'] = trim(fgets($auth_details));
                $_SESSION['user_id'] = $config['admin_user'];
                fclose($auth_details);
            }
        }
        return $config;
    }`

Statement
`$_SESSION['user_id'] = $config['admin_user'];`

ensures that a login is never requested

**Reply by billz** [reply]
@FranziHH perhaps you'd like to contribute a PR?

**Reply by jsantalo** [reply]
Correct, issue seems to be solved after commenting the abovementioned line.

**Reply by billz** [reply]
This introduces a new bug, so it is not a complete solution.

**Comment by FireDVB** [comment]
where can I find the file raspap-webgui/src/RaspAP/Auth/HTTPAuth.php on the Raspberry Pi (RaspAP).

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsantalo** [reply]
It can be found where the raspap-webgui is installed. In my case (raspberry pi OS), it was located in /var/www/html/src/RaspAP/Auth/HTTPAuth.php .

**Comment by FireDVB** [comment]
Thanks


I'll take a look.  What would you have to change to make it work again?

Sorry for my english.

Edit:
I took Httpauth.php from version 3.0.7 and overwrote it on the Pi.
Rights set to 755. Restarted the Pi but no change when there is a password in the raspap.auth which I entered via the web interface.

________________________________

**Comment by FireDVB** [comment]
I've solved it temporarily like this for now.

I entered my generated password in the HTTPAuth.php file and deleted the raspap.auth file then at least the login window with my specified password is accepted.

private $auth_default = array(
        'admin_user' => 'admin',
        'admin_pass' => 'generated password from raspap.auth enter her '

**Comment by jsantalo** [comment, marked as answer]
It seems v3.0.9 solves the issue thanks to @billz 👏 

Can you confirm it?
